mne/io/egi/egimff.py

```diff
@@ -0,0 +1,377 @@
+    info['dtype'], info['orig_format'] = {2: ('>i2', 'short'),
+                                          4: ('>f4', 'float'),
+                                          6: ('>f8', 'double')}[precision]
+    info['dtype'] = np.dtype(info['dtype'])
+    return info, mff_hdr
+
+
+def _read_events(input_fname, hdr, info):
+    """Read events for the record.
+
+    in:
+        input_fname : str with the file path
+        hdr : dictionary with the headers get
+              by read_mff_header
+        info : header info array
+    """
+    mff_events, event_codes = read_mff_events(input_fname, hdr)
+    info['n_events'] = len(event_codes)
+    info['event_codes'] = np.asarray(event_codes).astype('<U4')
+    events = np.zeros([info['n_events'],
+                      info['n_segments'] * info['n_samples']])
+    for n, event in enumerate(event_codes):
+        for i in mff_events[event]:
+            events[n][i] = 1.4012984643248171e-45
+    return events, info
+
+
+def _combine_triggers_mff(data, remapping=None):
+    """Combine binary triggers."""
+    new_trigger = np.zeros(data.shape[1])
+    if data.astype(bool).sum(axis=0).max() > 1:  # ensure no overlaps
+        logger.info('    Found multiple events at the same time '
+                    'sample. Cannot create trigger channel.')
+        return
+    if remapping is None:
+        remapping = np.arange(data) + 1
+    for d, event_id in zip(data, remapping):
+        idx = d.nonzero()
+        if np.any(idx):
+            new_trigger[idx] += event_id
+    return new_trigger
+
+
+@verbose
+def read_raw_egi_mff(input_fname, montage=None, eog=None, misc=None,
+                     include=None, exclude=None, preload=False, kind='raw',
+                     verbose=None):
+    """Read EGI mff binary as raw object.
+
+    .. note:: The trigger channel names are based on the
+              arbitrary user dependent event codes used. However this
+              function will attempt to generate a synthetic trigger channel
+              named ``STI 014`` in accordance with the general
+              Neuromag / MNE naming pattern.
+
+              The event_id assignment equals
+              ``np.arange(n_events - n_excluded) + 1``. The resulting
+              `event_id` mapping is stored as attribute to the resulting
+              raw object but will be ignored when saving to a fiff.
+              Note. The trigger channel is artificially constructed based
+              on timestamps received by the Netstation. As a consequence,
+              triggers have only short durations.
+
+              This step will fail if events are not mutually exclusive.
+
+    Parameters
+    ----------
+    input_fname : str
+        Path to the raw file.
+    montage : str | None | instance of montage
+        Path or instance of montage containing electrode positions.
+        If None, sensor locations are (0,0,0). See the documentation of
+        :func:`mne.channels.read_montage` for more information.
```

Comment:
I think we should stop putting these in the constructor and have people use the methods instead